src/snowflake/snowpark/functions.py

```diff
@@ -6582,8 +6582,16 @@ def to_array(e: ColumnOrName) -> Column:
         >>> df.select(to_array(col('a')).as_('ans')).collect()
         [Row(ANS='[\\n  1,\\n  2,\\n  3\\n]'), Row(ANS=None)]
     """
-    c = _to_col_if_str(e, "to_array")
-    return builtin("to_array")(c)
+    flattened_cols = []
+    for col in cols:
+        if isinstance(col, list):
+            flattened_cols.extend(col)
+        else:
+            flattened_cols.append(col)
```

Comment:
lets have a well defined API behavior here. Our goal is to accept the following:
```
to_array(col1, col2, col3)
to_array([col1, col2, col3])
```

let's not extend this to 
```
to_array([col1, col2], [col3], [col4])
```
It does not make much sense to support this.

we can do something like this

```
if len(cols) == 1 and isinstance(cols[0], list):
    cols = cols[0]
```